e2e/helpers/auth-helpers.ts

```diff
@@ -0,0 +1,37 @@
+import { type Page } from "@playwright/test";
+
+export class AuthHelpers {
+  constructor(public readonly page: Page) {}
+
+  async loginLocal() {
+    await this.page.goto("/");
+    await this.page.getByRole("textbox", { name: "Username" }).fill("admin");
+    await this.page
+      .getByRole("textbox", { name: "Password" })
+      .fill("password1");
+    await this.page
+      .getByRole("button", { name: "Log in to the dashboard" })
+      .click();
+  }
+
+  async loginCandid() {
+    await this.page.goto("/");
+    const popupPromise = this.page.waitForEvent("popup");
+    await this.page
+      .getByRole("link", { name: "Log in to the dashboard" })
+      .click();
+    const popup = await popupPromise;
+    await popup.getByRole("link", { name: "static" }).click();
+    await popup.getByRole("textbox", { name: "Username" }).fill("user1");
+    await popup.getByRole("textbox", { name: "Password" }).fill("password1");
+    await popup.getByRole("button", { name: "Login" }).click();
+  }
+
+  async login() {
+    let login = this.loginLocal;
+    if (process.env.AUTH_MODE === "candid") {
+      login = this.loginCandid;
+    }
+    await login();
```

Comment:
I think we can just call the functions above:

```suggestion
    if (process.env.AUTH_MODE === "candid") {
      await this.loginCandid();
    } else {
      await this.loginLocal();
    }
```

Reply:
Done.